src/calendar/index.tsx

```diff
@@ -91,31 +103,75 @@ const DEFAULT_WEEKDAYS: ShortLong<typeof commonBundle.messages>[] = [
 	{ short: 'satShort', long: 'saturday' }
 ];
 
-const factory = create({ icache, i18n, theme, focus }).properties<CalendarProperties>();
+const factory = create({
+	icache: createICacheMiddleware<CalendarIcache>(),
+	i18n,
+	theme,
+	focus
+}).properties<CalendarProperties>();
 
 export const Calendar = factory(function Calendar({
 	middleware: { icache, i18n, theme, focus },
-	properties
+	properties,
+	id
 }) {
 	const themeCss = theme.classes(css);
 	const { messages: commonMessages } = i18n.localize(commonBundle);
-	const callDateFocus = icache.getOrSet('callDateFocus', false);
-	const defaultDate = icache.getOrSet('defaultDate', new Date());
-	const focusedDay = icache.getOrSet('focusedDay', 1);
-	const monthLabelId = icache.getOrSet('monthLabelId', uuid());
-	const popupOpen = icache.getOrSet('popupOpen', false);
-	const shouldFocus = focus.shouldFocus();
 
 	const {
 		labels = i18n.localize(calendarBundle).messages,
 		aria = {},
-		selectedDate,
 		minDate,
 		maxDate,
+		initialValue,
 		weekdayNames = getWeekdays(commonMessages),
 		firstDayOfWeek = 0
 	} = properties();
-	const { year, month } = getMonthYear();
+
+	const defaultDate = initialValue || new Date();
+	const {
+		initialYear = defaultDate.getFullYear(),
+		initialMonth = defaultDate.getMonth()
+	} = properties();
+	const existingInitialValue = icache.get('initialValue');
+	const existingInitialMonth = icache.get('initialMonth');
+	const existingInitialYear = icache.get('initialYear');
+	const callDateFocus = icache.getOrSet('callDateFocus', false);
+	const focusedDay = icache.getOrSet('focusedDay', 1);
+	const monthLabelId = icache.getOrSet('monthLabelId', id);
+	const popupOpen = icache.getOrSet('popupOpen', false);
+	const shouldFocus = focus.shouldFocus();
+	let value = icache.get('value') || new Date();
+
+	if (!initialValue && !existingInitialValue) {
+		let value = toDate(defaultDate);
```

Comment:
Do we need this `let` or should it change the parent `value`?

Reply:
It should change the existing value. Updated 👍